### PR TITLE
solution

### PR DESCRIPTION
**Pull request by GoldenSpade**

### CODE REVIEW

**Review by stas-dukach — APPROVED** [review]
good

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,13 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => century
+    ? person.sex === 'm' && Math.ceil(person.died / 100) === century
```
Comment (aionova) [inline src/averageAges.js:19]: what if we try not to repeat `person.sex === 'm'`?

File: src/averageAges.js [hunk 2]
```diff
@@ -37,7 +39,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => withChildren
+    ? person.sex === 'f' && people.some(child => child.mother === person.name)
+    : person.sex === 'f'
```
Comment (aionova) [inline src/averageAges.js:44]: try not to repeat `person.sex === 'f'`?

**Review by aionova — APPROVED** [review]
Looks good, but could be refactored

**Pull request closed without merging**